src/io/flutter/module/FlutterModuleBuilder.java

```diff
@@ -140,4 +153,52 @@ public boolean validate() throws ConfigurationException {
     public void dispose() {
     }
   }
+
+  // Wraps standard project settings to add project name validation.
+  private static class FlutterProjectSettingsStep extends ModuleWizardStep {
+
+    private final ProjectSettingsStep mySettingsStep;
+
+    FlutterProjectSettingsStep(ProjectSettingsStep settingsStep) {
+      mySettingsStep = settingsStep;
+    }
+
+    @Override
+    public JComponent getComponent() {
+      return mySettingsStep.getComponent();
+    }
+
+    @Override
+    public void updateDataModel() {
+      // Done during validation.
+    }
+
+    @Override
+    public boolean isStepVisible() {
+      return mySettingsStep.isStepVisible();
+    }
+
+    @Override
+    public boolean validate() throws ConfigurationException {
+
+      mySettingsStep.updateDataModel();
+
+      final WizardContext context = mySettingsStep.getContext();
+      final String projectName = context.getProjectName();
+
+      if (!FlutterUtils.isValidDartdentifier(projectName)) {
+        throw new ConfigurationException("Invalid Flutter project name '" + projectName + "'; must be a valid Dart identifier.");
+      }
+
+      if (projectName.length() > FlutterConstants.MAX_PROJECT_NAME_LENGTH) {
+        throw new ConfigurationException("Flutter project name '" +
+                                         projectName +
```

Comment:
We may not want to display the project name here (given that it will be pretty long).